### PR TITLE
Add howto for applying a cowboy on qastaging

### PR DESCRIPTION
**Pull request by jugmac00**

### CODE REVIEW

File: how-to/applying-a-cowboy-to-qastaging.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    circumstances.
+
+Before you apply a cowboy, please have a quick conversation with your team
+where the reason for the cowboy is thoroughly discussed.
+
+Applying the cowboy
+===================
+
+#. SSH into bastion ``ssh launchpad-bastion-ps5`` and switch to the
+   ``stb-vbuilder-qastaging`` user by running
```
Comment (ineesalmeida) [inline how-to/applying-a-cowboy-to-qastaging.rst:35]: Looking at the rest of the stuff below, I'm guessing this should be `stg-launchpad`? Same in the line below this one

File: how-to/applying-a-cowboy-to-qastaging.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+   currently applied git commit id.
+
+#. Checkout that commit id locally.
+
+#. Apply the planned changes to your local checkout and create a local patch
+   file via e.g. ``git diff > <description>.patch``.
+
+#. Log into the appropriate unit where you want to apply the patch to.
+
+    - The following steps describe the process of applying a patch to the
```
Comment (ineesalmeida) [inline how-to/applying-a-cowboy-to-qastaging.rst:50]: I'd remove this bullet point and just add these details to the sentence above, example:
```
Log into the appropriate unit where you want to apply the patch to. In this example, we will apply the patch to the application server (``launchpad-appserver``).
```

File: how-to/applying-a-cowboy-to-qastaging.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      each of the units.
+
+    - ssh into one of the units via ``juju ssh launchpad-appserver/2``
+
+    - Change into the directory with the source code via
+      ``cd /srv/launchpad/code``.
+
+    - Apply the patch via ``git apply <description>.patch``.
+
+    - Restart the system via ``sudo systemctl restart launchpad``.
```
Comment (ineesalmeida) [inline how-to/applying-a-cowboy-to-qastaging.rst:65]: Is it worth mentioning that some cowboys to some units don't require this restart?
If we are changing something that runs from a cronscript there is no need to restart a service for example.

Maybe we can leave this as it shouldn't hurt anyways

**Review by ineesalmeida — APPROVED** [review]
Looks great, thank you for adding this! Just a couple of comments to be addressed

File: how-to/applying-a-cowboy-to-qastaging.rst (hunk 3 above)
Comment (jugmac00) [reply how-to/applying-a-cowboy-to-qastaging.rst:65]: Thanks! I combined this info with your suggestion on how to improve the introduction.

File: how-to/applying-a-cowboy-to-qastaging.rst (hunk 2 above)
Comment (jugmac00) [reply how-to/applying-a-cowboy-to-qastaging.rst:50]: Thanks - that improves readability a lot!

File: how-to/applying-a-cowboy-to-qastaging.rst (hunk 1 above)
Comment (jugmac00) [reply how-to/applying-a-cowboy-to-qastaging.rst:35]: Great catch. I copied this part from another how-to.. it feels very repetitive to add this kind of information so often, but I also do not like to leave it out, as logging into the bastion might be something everybody is familiar, especially when only joined recently.

**Comment by jugmac00** [comment]
@ineesalmeida Thanks for the review! On point as usual!

**Pull request merged**